### ISSUE TITLE
urllib.parse.Misimplementation of urljoin

### ISSUE THREAD

**Issue by SBlechmann** [RWTH-EBC/FiLiP#360]
**Describe the bug**
The urllib.parse.urljoin, e.g. when using orion base url and the version endpoint in the get_version() function, works a little different than originally expected.

urljoin(url1, url2)

E.g. if orion is running behind "example.org/orion" and I run the get_version, the function will end up querying "example.org/version".
This is because urljoin does not simply concatenate the URLs but cuts the first url depending on how many "/" are part of second url.

**To Reproduce**
see above, give the orion client a url with a path /orion

**Expected behavior**
Get version; instead getting nothing cause url is wrong.

**Comment by tstorek** [comment]
@djs0109 The problem persists because the ngsi_version contains a '/' leading wrong concatenation. of the url. 
This should be fixed soon for the other urls if "example.org/orion/" is used. Alternatively, the version check must be adapted

**Comment by djs0109** [comment]
@tstorek good point. I will fix the version enumeration, i.e. remove the starting `/` of the versions
```python
class NgsiURLVersion(str, Enum):
    """
    URL part that defines the NGSI version for the API.
    """

    v2_url = "v2"
    ld_url = "ngsi-ld/v1"
```

**Comment by tstorek** [comment]
@djs0109 thanks, at the same time it would be good to validate the baserul. Maybe using pydantics AnyHttpUrl. As I understand that one will always add the trailing slash if missing
